api.go

```diff
@@ -200,7 +202,217 @@ func (r *Router) RouterCallImpl(ctx context.Context,
 	}
 }
 
-// todo: router_map_callrw
+type MapCallMetaData struct {
+	Resp  []interface{}
+	Error error
+
+	replicaset *Replicaset
+	future     *tarantool.Future
+}
+
+func (r *Router) getMapCallSelectCases(req tarantool.Request) []reflect.SelectCase {
+	metaDatas := make([]<-chan *MapCallMetaData, 0)
+	for _, replicaset := range r.idToReplicaset {
```

Comment:
i think current logic looks  too hard and weight, in my mind -  the best way - to use just a slice with meta info; then u can send slice index  in range and ref to rs to goroutine with calll and return structs instead strange reflect struct